packages/labs/task/src/test/task_test.ts

```diff
@@ -222,6 +231,32 @@ suite('Task', () => {
     assert.equal(el.taskValue, `error`);
   });
 
+  test('task functions receive an AbortSignal', async () => {
+    const el = getTestElement({args: () => [el.a, el.b], autoRun: false});
+    await renderElement(el);
+
+    // Initially we have no signal because the task function hasn't been run
+    await tasksUpdateComplete();
+    assert.equal(el.task.status, TaskStatus.INITIAL);
+    assert.equal(el.signal, undefined);
+
+    // When the task is run, we'll get a signal
+    el.task.run();
+    assert.equal(el.task.status, TaskStatus.PENDING);
+    assert.ok(el.signal);
+
+    // If we start a new run before the previous is complete, the signal
+    // should be aborted
+    const previousSignal = el.signal;
+    el.task.run();
+    assert.equal(el.task.status, TaskStatus.PENDING);
+    assert.strictEqual(previousSignal?.aborted, true);
+
+    // And the new run should have a fresh, unaborted, signal
+    assert.notStrictEqual(previousSignal, el.signal);
+    assert.strictEqual(el.signal?.aborted, false);
```

Comment:
let's also assert that if this new task completes, and then we start another task afterwards, the completed task's signal isn't aborted